apps/passport-server/src/services/poapService.ts

```diff
@@ -295,6 +297,67 @@ export class PoapService {
     });
   }
 
+  /**
+   * Validates that a serialized ZKEdDSAEventTicketPCD is a valid
+   * Edge City Denver 7-day ticket and returns the ID of that ticket.
+   *
+   * This function throws an error in the case that the PCD is not
+   * valid; for example, here are a few invalid cases
+   *  1. Wrong PCD type
+   *  2. Wrong EdDSA public key
+   *  3. PCD proof is invalid
+   *  4. Event of ticket is not Edge City Denver
+   *  5. Type of ticket is not 7-day pass
+   */
+  private async validateEdgeCityDenverTicket(
+    serializedPCD: string
+  ): Promise<string> {
+    return traced("poap", "validateEdgeCityDenverTicket", async (span) => {
+      if (!process.env.GENERIC_ISSUANCE_EDDSA_PRIVATE_KEY)
+        throw new Error(
+          "Missing generic issuance eddsa private key .env value"
+        );
+      const pcd = await this.validateZKEdDSAEventTicketPCD(
+        serializedPCD,
+        await getEdDSAPublicKey(process.env.GENERIC_ISSUANCE_EDDSA_PRIVATE_KEY)
```

Comment:
ideally the public key would also be an environment variable in zupass, 10% sure it already is